### PR TITLE
Suggest contributors to create a new branch to work on a new issue

### PR DESCRIPTION
**Pull request by sks444**
Closes https://github.com/pythonindia/junction/issues/628 https://github.com/pythonindia/junction/issues/635

### CODE REVIEW

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -26,13 +26,16 @@ land a single large feature.
 
 Checklist:
 
-1. All pull requests *must* be made against the ``master`` branch.
-2. Include tests for any functionality you implement. Contributions that
+1. Always create a new branch to work on a new issue.
```
Comment (sks444) [inline .github/CONTRIBUTING.rst]: I was also thinking to mention the naming convention of the branch that would created as in the name should be related to the changes you're doing. Also we could add a command to show how to create a new branch.

But felt like those information would be redundant, suggestions?

File: .github/CONTRIBUTING.rst [hunk 2]
```diff
@@ -26,13 +26,16 @@ land a single large feature.
...
+3. All pull requests *must* be made against the ``master`` branch.
+4. Include tests for any functionality you implement. Contributions that
    improve existing tests are welcome.
-3. Update documentation as necessary and provide documentation for any new
+5. Update documentation as necessary and provide documentation for any new
    functionality.
-4. In case of UI changes, please include screenshots.
-5. Add yourself to ``CONTRIBUTORS.txt`` if you're not there already. :)
+6. In case of UI changes, please include screenshots.
+7. Add yourself to ``CONTRIBUTORS.txt`` if you're not there already. :)
```
Comment (sks444) [inline .github/CONTRIBUTING.rst]: I couldn't find the file ``CONTRIBUTORS.txt``,  so created an [issue](https://github.com/pythonindia/junction/issues/635) regarding it. Let me know the issue is valid, I could make the changes related to that in this pr only.

Comment (pradyunsg) [reply .github/CONTRIBUTING.rst]: Yea, go ahead and remove this line.

I plan on revamping the README at a later date.

Comment (sks444) [reply .github/CONTRIBUTING.rst]: Done.

File: .github/CONTRIBUTING.rst (hunk 1 above)
Comment (ananyo2012) [reply .github/CONTRIBUTING.rst]: Yes you can add the command `git checkout -b <branch-name>` . Helpful for new comers. We should actually add a getting started with git link somwhere.

Comment (sks444) [reply .github/CONTRIBUTING.rst]: Added the command.

@ananyo2012 Let me know a good git getting started guide that you prefer, I will try to plug it in as note before the `Checklist`

File: .github/CONTRIBUTING.rst [hunk 3]
```diff
@@ -23,16 +23,20 @@ Pull Requests should be small to facilitate easier review. Keep them
...
-
 Checklist:
 
-1. All pull requests *must* be made against the ``master`` branch.
-2. Include tests for any functionality you implement. Contributions that
+1. Always create a new branch to work on a new issue::
+
+    $ git checkout -b <branch-name>
+
+2. Make sure your branch is rebased with upstream master before you file
```
Comment (pradyunsg) [inline .github/CONTRIBUTING.rst]:
```suggestion
2. Make sure your branch is up-to-date with upstream master before you file
```

**Comment by pradyunsg** [comment]
Thanks @sks444! ^>^

**Pull request merged**